android/src/main/java/io/ably/flutter/plugin/AblyMessageCodec.java

```diff
@@ -82,9 +86,11 @@ T decode(Map<String, Object> jsonMap) {
 
   private Map<Byte, CodecPair> codecMap;
   private static final Gson gson = new Gson();
+  private final CipherParamsStorage cipherParamsStorage;
 
-  public AblyMessageCodec() {
+  public AblyMessageCodec(CipherParamsStorage cipherParamsStorage) {
     final AblyMessageCodec self = this;
+    this.cipherParamsStorage = cipherParamsStorage;
     codecMap = new HashMap<Byte, CodecPair>() {
```

Comment:
This can also be done in `AblyMethodCallHandler.java`, doing this in this PR would add a lot of unrelated changes. Having said that, I don't want to create an issue for this yet because I am not sure of the solution: It could just be a switch statement inside `onMethodCall`, instead of using this map at all: 

https://github.com/ably/ably-flutter/blob/0898c07cf4487b8b8614260ad28daa11e64f336d/android/src/main/java/io/ably/flutter/plugin/AblyMethodCallHandler.java#L153-L165

There are more places where things can go wrong in the current implementation, and reducing code in this hot path would be useful.

Reply:
Oh, please see https://github.com/ably/ably-flutter/issues/180, I made one in September 😁

Reply:
I made this comment only because the constructor code was massive and extracting the building to a method should not make any change to the code. It wasn't a comment on your implementation

Reply:
I agree it is big. However, extracting it would be quite a large change in the diff and would like to keep it for a separate PR. If I extracted it, it would be less clear that I added:
```
put(PlatformConstants.CodecTypes.cipherParams,
            new CodecPair<>(self::encodeCipherParams, self::decodeCipherParams));
```

Reply:
Ok then, I'll leave this with you 👍